tools/python/util/convert_onnx_models_to_ort.py

```diff
@@ -239,13 +229,46 @@ def convert_onnx_models_to_ort():
     else:
         session_options_config_entries["session.qdqisint8allowed"] = "0"
 
-    for optimization_level in args.optimization_level:
-        print(f"Converting models and creating configuration file for optimization level '{optimization_level}'")
-        _convert(model_path_or_dir, optimization_level, args.use_nnapi, args.use_coreml, custom_op_library,
-                 args.save_optimized_onnx_model, args.allow_conversion_failures, args.target_platform,
-                 session_options_config_entries)
-
-        _create_config_file_from_ort_models(model_path_or_dir, optimization_level, args.enable_type_reduction)
+    for optimization_style in optimization_styles:
+        print("Converting models with optimization style '{}' and level '{}'".format(
+            optimization_style.name, optimization_level_str))
+
+        converted_models = _convert(
+            model_path_or_dir=model_path_or_dir, output_dir=None,
+            optimization_level_str=optimization_level_str, optimization_style=optimization_style,
+            custom_op_library=custom_op_library,
+            create_optimized_onnx_model=args.save_optimized_onnx_model,
+            allow_conversion_failures=args.allow_conversion_failures,
+            target_platform=args.target_platform,
+            session_options_config_entries=session_options_config_entries)
+
+        with contextlib.ExitStack() as context_stack:
+            if optimization_style == OptimizationStyle.Runtime:
+                # Convert models again without runtime optimizations.
+                # Runtime optimizations may not end up being applied, so we need to use both converted models with and
+                # without runtime optimizations to get a complete set of ops that may be needed for the config file.
+                model_dir = model_path_or_dir if model_path_or_dir.is_dir() else model_path_or_dir.parent
```

Comment:
In order to get the complete set of ops do we need to make a change to inference_session.cc to have a notion of optimizing but just with the optimizers that would be used to create the saved optimizations? 

i.e. you want [https://github.com/microsoft/onnxruntime/blob/d478a53d433473cad296bcdae2edc43a0fef210c/onnxruntime/core/session/inference_session.cc#L2378](https://github.com/microsoft/onnxruntime/blob/d478a53d433473cad296bcdae2edc43a0fef210c/onnxruntime/core/session/inference_session.cc#L2378) to be used, but instead of a save context with the SatDirectApplicationContext. 

If you don't limit the optimizers there may be some other level 2 or level 3 ones that can apply that aren't supported by runtime optimizations, resulting in different ops in the final model.